snuba/datasets/errors_replacer.py

```diff
@@ -393,20 +393,39 @@ def process_tombstone_events(
         # not.
         return None
 
-    # XXX: We need to construct a query that works on both event_id columns,
-    # either represented as UUID or as hyphenless FixedString. That's why we
-    # use replaceAll(toString()).
-    where = """\
-        PREWHERE replaceAll(toString(event_id), '-', '') IN (%(event_ids)s)
+    from_ts = message.get("from_timestamp")
+    to_ts = message.get("to_timestamp")
+    if from_ts is not None and to_ts is not None:
+        from_ts = datetime.strptime(from_ts, settings.PAYLOAD_DATETIME_FORMAT)
+        to_ts = datetime.strptime(to_ts, settings.PAYLOAD_DATETIME_FORMAT)
+        ts_condition = f"""
+        AND timestamp >= toDateTime('{from_ts.strftime(DATETIME_FORMAT)}') AND
+            timestamp < toDateTime('{to_ts.strftime(DATETIME_FORMAT)}')
+        """
+    else:
+        ts_condition = ""
+
+    if state_name == ReplacerState.EVENTS:
+        event_id_lhs = "cityHash64(toString(event_id))"
+        event_id_list = ", ".join(
+            [
+                f"cityHash64('{str(uuid.UUID(event_id)).replace('-', '')}')"
+                for event_id in event_ids
+            ]
+        )
+    else:
+        event_id_lhs = "event_id"
```

Comment:
errors also has cityHash64 in the primary key so could benefit from same strategy

Reply:
fixed